runtime/lua/vim/treesitter.lua

```diff
@@ -25,12 +25,12 @@ setmetatable(M, {
  })
 
 --- Creates a new parser.
---
--- It is not recommended to use this, use vim.treesitter.get_parser() instead.
---
--- @param bufnr The buffer the parser will be tied to
--- @param lang The language of the parser
--- @param opts Options to pass to the language tree
+---
+--- It is not recommended to use this, use vim.treesitter.get_parser() instead.
+---
+--- @param bufnr The buffer the parser will be tied to
+--- @param lang The language of the parser
+--- @param opts Options to pass to the created language tree
 function M._create_parser(bufnr, lang, opts)
```

Comment:
This isn't that big of a deal, but when I was first trying to read through the source code, I found it confusing that `{create, get}_parser` actually returns a languagetree, which may have multiple parsers. Maybe in the future, a better distinction between language trees and treesitter parsers could be added to the docs.

Reply:
Indeed this is confusing... We will deprecate those functions in a later pr then, and have function like `setup` to simplify things.

Reply:
An alternative would be to rename the language tree to multilanguageparser/parsertree. The language trees have a lot of name clashes with the parser trees they produce. `get_parser()` could maybe be `get_root_parser()` `get_top_level_parser()` `get_parser_tree`. The languagetree still implements the parser API it replaced.